### PR TITLE
Fix broken issue link when a group bug is used + add tests.

### PR DESCRIPTION
**Pull request by Dor1s**
Noticed on https://clusterfuzz.com/testcase-detail/5103503034417152, the issue link points to `https://clusterfuzz.com/testcase-detail/null`.

### CODE REVIEW

**Comment by Dor1s** [comment]
Wondering whether `get_issue_for_testcase` also should check `group_bug_information`. If yes, I'll create a helper for getting issue id and use it in both places.

**Comment by Dor1s** [comment]
some network glitch, trying again

**Comment by Dor1s** [comment]
> Wondering whether get_issue_for_testcase also should check group_bug_information. If yes, I'll create a helper for getting issue id and use it in both places.

After taking a closer look at the call sites for `get_issue_for_testcase`, it seems to be that it shouldn't hurt to update it as well. I'll upload another patchset.

**Comment by inferno-chromium** [comment]
@oliverchang should review this. i think i did a wrong review here, since get_issue_for_url seems like the wrong place. Either get_issue_for_testcase as you mentioned or somewhere else. but i just worry that group bug information might cause wrong result elsewhere. since @oliverchang refactored this, he will know this inside out.

**Comment by Dor1s** [comment]
Ack, waiting for Oliver to check what's going on first.

**Comment by Dor1s** [comment]
Sorry I couldn't resist and made that change, as it actually made the whole CL smaller and likely better.

**Comment by Dor1s** [comment]
Thanks for clarifying! I've reverted that change and added a comment to `get_issue_for_testcase`.

File: src/appengine/libs/issue_management/issue_tracker_utils.py [hunk 1]
```diff
@@ -154,10 +156,17 @@ def get_similar_issues_url(issue_tracker, testcase, only_open=True):
 def get_issue_url(testcase):
   """Return issue url for a testcase."""
```
Comment (inferno-chromium) [inline src/appengine/libs/issue_management/issue_tracker_utils.py]: Maybe expand on description here that this is for display and account for group bug.

Comment (Dor1s) [reply src/appengine/libs/issue_management/issue_tracker_utils.py]: Done!

**Pull request merged**